### DISCUSSION TITLE
About first interface IP address of router

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#240] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
https://github.com/ANLAB-KAIST/KENSv3/discussions/90  according to this discussion, I can use GetIpAddr() to figure out which source IP to use when destination IP is given. But in the implementation of OSPF protocol, router is identified by first IP address of its interfaces.
I don't know how to figure out this IP, since in this case, destination IP is not clear. How can I?

**Comment by Pusnow** [comment, marked as answer]
Use "getIPAddr(0); // retrieve 0th interface's IP"

**Reply by 03psb** [reply]
Thank you for your answer!
